Sources/SwifterSwift/SwiftStdlib/DefaultStringInterpolationExtensions.swift

```diff
@@ -0,0 +1,46 @@
+// DefaultStringInterpolationExtensions.swift - Copyright 2023 SwifterSwift
+
+#if canImport(Foundation)
+import Foundation
+#endif
+
+public extension DefaultStringInterpolation {
+    /// SwifterSwift: Interpolates the given value's textual representation
+    /// into the string literal being created if the value is not `nil` and the
+    /// optional predicate returns `true`, otherwise append a placeholder. 
+    ///
+    /// Do not call this method directly. It is used by the compiler when
+    /// interpreting string interpolations. Instead, use string
+    /// interpolation to create a new string by including values, literals,
+    /// variables, or expressions enclosed in parentheses, prefixed by a
+    /// backslash (`\(`...`, placeholder: `...`)`).
+    ///
+    ///     var token: Int? = nil
+    ///     print("\(token, placeholder: "-")")
+    ///     // Prints "-"
+    ///     token = 0
+    ///     print("\(token, placeholder: "-")")
+    ///     // Prints "0"
+    ///     print("\(token, placeholder: "-", predicate: { $0 > 0} )")
+    ///     // Prints "-"
+    ///
+    /// - Parameters:
+    ///   - value: The values, literals, variables, or expressions to be interpolated.
+    ///   - placeholder: The string that displays when `value` is `nil`.
+    ///   - predicate: A closure that takes unwrapped `value` as its argument and returns `true` if the `value`'s
+    ///                textual representation  should be interpolated into the string literal or `false` if the
+    ///                `placeholder` should be interpolated into the string literal. You can use a `nil` value for
+    ///                this parameter.
+    mutating func appendInterpolation<T>(_ value: T?,
+                                         placeholder: @autoclosure () -> String,
+                                         predicate: ((T) throws -> Bool)? = nil) rethrows
+    {
+        switch value {
+        case .some(let value) where try predicate?(value) != false:
```

Comment:
Why not true evaluation?
```suggestion
        case .some(let value) where try predicate?(value) == true:
```

Reply:
It may be `nil`, it's a little trick to avoid writing `predicate?(value) == nil || predicate?(value) == true`.

Reply:
But If is `nil` it wouldn't be also `true` right?

Reply:
My intention is append the literal of value if
1. the given `value` is not `nil` and `predicate` is not provided
or
2. the given `value` is not `nil` and `predicate` is provided  and the test result of the value is `true`

The following code presents the logic:

```swift
case .some(let value) where (predicate == nil || try predicate?(value) == true):
            appendLiteral("\(value)")
```

`predicate?(value) != false` is just an equivalence to `predicate == nil || try predicate?(value) == true`

Reply:
Another equivalence is

```swift
        case .some(let value) where try predicate?(value) ?? true:
```

do you think it's more easy to read?

Reply:
Ah no makes sense